src/base-importer/BaseImporter.js

```diff
@@ -0,0 +1,358 @@
+        }
+
+        // split around special characters
+        const re = /([^\x00-\x1f]+)|([\x00-\x1f]+)/gm
+        let components = []
+        let m
+        while ((m = re.exec(string)) !== null) {
+            if (m[1]) {
+                components.push(m[1])
+            }
+            else {
+                components.push(this._escapeSequenceDynamicValue(m[2]))
+            }
+        }
+        return new DynamicString(...components)
+    }
+
+    _createPawRequest(context, request) {
+        return context.createRequest(
+            request.get('name'),
+            request.get('method'),
+            this._toDynamicString(request.get('url'), true, true),
+        )
+    }
+
+    _setHeaders(pawReq, headers) {
+        headers.forEach((value, key) => {
+            pawReq.setHeader(
+                this._toDynamicString(key, true, true),
+                this._toDynamicString(value, true, true)
+            )
+        })
+        return pawReq
+    }
+
+    /*
+        TODO: Add OAuth1 support when API-flow will support it
+    */
+    _setAuth(pawReq, auth) {
+        if (auth) {
+            if (auth instanceof Auth.Basic) {
+                const dv = new DynamicValue(
+                    'com.luckymarmot.BasicAuthDynamicValue',
+                    {
+                        username: auth.get('username', ''),
+                        password: auth.get('password', '')
+                    }
+                )
+                pawReq.setHeader('Authorization', new DynamicString(dv))
+            }
+            else if (auth instanceof Auth.OAuth2) {
+                const dv = new DynamicValue(
+                    'com.luckymarmot.OAuth2DynamicValue',
+                    {
+                        grant_clitype: auth.get('flow', ''),
+                        authorization_uri: auth.get('authorizationUrl', ''),
+                        access_token_uri: auth.get('tokenUrl', ''),
+                        scope: (auth.get('scopes') || []).join(' ')
+                    }
+                )
+                pawReq.setHeader('Authorization', new DynamicString(dv))
+            }
+            else {
+                /* eslint-disable no-console */
+                console.error(
+                    'Auth type ' +
+                    auth.get('type') +
+                    ' is not supported in Paw'
+                )
+                /* eslint-enable no-console */
+            }
+        }
+        return pawReq
+    }
+
+    _setFormDataBody(pawReq, body) {
+        const keyValues = body.map((value, key) => {
+            return [
+                this._toDynamicString(key, true, true),
+                this._toDynamicString(value, true, true),
+                true
+            ]
+        }).toArray()
+        const dv = new DynamicValue(
+            'com.luckymarmot.BodyMultipartFormDataDynamicValue', {
+                keyValues: keyValues
+            }
+        )
+        pawReq.body = new DynamicString(dv)
```

Comment:
We should be using here instead the setter: `request.multipartBody`

```javascript
multipartBody = {
  key: this._toDynamicString(value, true, true)
}
```

Sure, it doesn't allow the keys to contain dynamic values, but I don't think it's the case at the moment.

Otherwise, if we don't use this *"simplified"* JS API, it will break in 2.4.